### PR TITLE
DDP-4605 add zip code check to Verify Address API

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

When Verify Address API is called and `studyGuid` is provided, we also check study rules for any potential warnings. We return warnings for both the "entered" and "suggested" addresses.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/docs/specification/src/endpoints/user.profile.addresses.verify.yml [hunk 1]
```diff
@@ -12,9 +12,62 @@ post:
...
+                  description: if provided, will check address against study rules for any potential issues
   responses:
     200:
-      $ref: '../pepper.yml#/components/responses/MailingAddress'
+      description: a corrected address that's better suited for deliverability, along with any warnings
+      content:
+        application/json:
+          schema:
+            allOf:
+              - $ref: '../pepper.yml#/components/schemas/BaseMailAddress'
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/endpoints/user.profile.addresses.verify.yml:33]: We're not dealing with all the properties in a mailing address (e.g. `description`, `validationStatus`, etc) so it's just the basic address.

File: pepper-apis/docs/specification/src/endpoints/user.profile.addresses.verify.yml [hunk 2]
```diff
@@ -12,9 +12,62 @@ post:
...
+              - type: object
+                required:
+                  - studyGuid
+                  - warnings
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/endpoints/user.profile.addresses.verify.yml:37]: Looks something like this:

Screen Shot 2020-05-26 at 2 54 54 PM

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 3]
```diff
@@ -1755,26 +1740,103 @@ MailingAddress:
...
+      required:
+        - description
+        - name
+        - street1
+        - city
+        - state
+        - country
+        - zip
+        - phone
+      properties:
```
Comment (MocanaAtBroad) [inline pepper-apis/docs/specification/src/components/schemas.yml:1758]: We missing street2?

**Review by MocanaAtBroad — COMMENTED** [review]
Looks like what we discussed! LGTM

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 3 above)
Comment (yufengwng) [reply pepper-apis/docs/specification/src/components/schemas.yml:1758]: This was the original list of required properties from the `MailingAddress` object schema, which didn't have `street2`. I just left it as-is.

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**